lib/services/data/plan/plan_db_repository.dart

```diff
@@ -8,27 +8,39 @@ import 'package:fokus/model/db/plan/plan.dart';
 import 'package:fokus/services/data/db/db_repository.dart';
 
 mixin PlanDbRepository implements DbRepository {
+	Future<List<Plan>> getCaregiverPlans(ObjectId caregiverId, {bool activeOnly = true}) {
+		var query = _buildPlanQuery(caregiverId: caregiverId, activeOnly: activeOnly);
+		return dbClient.queryTyped(Collection.plan, query, (json) => Plan.fromJson(json));
+	}
+
 	Future<List<Plan>> getChildPlans(ObjectId childId, {bool activeOnly = true}) {
-		var query = _buildPlanQuery(childId, activeOnly: activeOnly);
+		var query = _buildPlanQuery(childId: childId, activeOnly: activeOnly);
 		return dbClient.queryTyped(Collection.plan, query, (json) => Plan.fromJson(json));
 	}
 
 	Future<List<Plan>> getChildPlanInstances(ObjectId childId, {ObjectId planId, Date date, bool activeOnly = false}) {
-		var query = _buildPlanQuery(childId, planId: planId, date: date, activeOnly: activeOnly);
+		var query = _buildPlanQuery(childId: childId, planId: planId, date: date, activeOnly: activeOnly);
 		return dbClient.queryTyped(Collection.planInstance, query, (json) => Plan.fromJson(json));
 	}
 
 	Future<PlanInstance> getActiveChildPlanInstance(ObjectId childId) {
-	  return dbClient.queryOneTyped(Collection.planInstance, _buildPlanQuery(childId, state: PlanInstanceState.active), (json) => PlanInstance.fromJson(json));
+		return dbClient.queryOneTyped(Collection.planInstance, _buildPlanQuery(childId: childId, state: PlanInstanceState.active), (json) => PlanInstance.fromJson(json));
 	}
 
 	Future<List<PlanInstance>> getPlanInstancesForPlans(ObjectId childId, List<ObjectId> planIDs) {
-		var query = _buildPlanQuery(childId).and(where.oneFrom('planID', planIDs));
+		var query = _buildPlanQuery(childId: childId).and(where.oneFrom('planID', planIDs));
 		return dbClient.queryTyped(Collection.planInstance, query, (json) => PlanInstance.fromJson(json));
 	}
 
-	SelectorBuilder _buildPlanQuery(ObjectId childId, {ObjectId planId, Date date, bool activeOnly = false, PlanInstanceState state}) {
-		var query = where.eq('assignedTo', childId);
+	SelectorBuilder _buildPlanQuery({ObjectId caregiverId, ObjectId childId, ObjectId planId, Date date, bool activeOnly = false, PlanInstanceState state}) {
+		var query;
+		if (childId != null)  {
+			query = where.eq('assignedTo', childId);
+			if (caregiverId != null)
+				query = query.and(where.eq('createdBy', caregiverId));
+		}
+		else if (caregiverId != null)
+			query = where.eq('createdBy', caregiverId);
 		if (activeOnly)
 			query.and(where.eq('active', true));
```

Comment:
I would do it like that:
```suggestion
		SelectorBuilder query;
		var addExpression = (expression) => query == null ? (query = expression) : query.and(expression);
		if (childId != null)
			addExpression(where.eq('assignedTo', childId));
		else if (caregiverId != null)
			addExpression(where.eq('createdBy', caregiverId));
		if (activeOnly)
			addExpression(where.eq('active', true));
```
and so on..

Reply:
👍